### PR TITLE
Add support for Windows devices

### PR DESCRIPTION
**Pull request by jonstoneman**
## Change list

Add `WindowsDriver` and `WindowsElement` for use with https://github.com/appium/appium-windows-driver
 
## Types of changes

What types of changes are you proposing/introducing to .NET client?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

##Details

Some methods throw NotImplementedExceptions due to https://github.com/Microsoft/WinAppDriver/ (and therefore https://github.com/appium/appium-windows-driver) not supporting the functionality.

### CODE REVIEW

**Comment by jlipps** [comment]
👍  however lots of merges in the git log here

**Comment by TikhomirovSergey** [comment]
Wow!
Will check it ASAP

**Comment by TikhomirovSergey** [comment]
@jonstoneman @yodurr 
Please provide licence header at new added files

File: appium-dotnet-driver/Appium/Windows/WindowsDriver.cs [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        /// Create a Windows Element
+        /// </summary>
+        /// <param name="elementId">element to create</param>
+        /// <returns>WindowsElement</returns>
+        protected override RemoteWebElement CreateElement(string elementId)
+        {
+            return new WindowsElement(this, elementId);
+        }
+
+        public override W ScrollTo(string text)
```
Comment (TikhomirovSergey) [inline appium-dotnet-driver/Appium/Windows/WindowsDriver.cs]: [There is the commit](https://github.com/appium/appium-dotnet-driver/commit/6c3dddf1bdb184a4524069c50e9ffe1a0e815103). Obsolete code was removed.
Please remove this method.

File: appium-dotnet-driver/Appium/Windows/WindowsDriver.cs [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+        {
+            throw new NotImplementedException();
+        }
+
+        public override W ScrollToExact(string text)
```
Comment (TikhomirovSergey) [inline appium-dotnet-driver/Appium/Windows/WindowsDriver.cs]: [There is the commit](https://github.com/appium/appium-dotnet-driver/commit/6c3dddf1bdb184a4524069c50e9ffe1a0e815103). Obsolete code was removed.
Please remove this method.

File: appium-dotnet-driver/appium-dotnet-driver.csproj [hunk 3]
```diff
@@ -67,6 +67,7 @@
     <Compile Include="Appium\Enums\MobileBrowserType.cs" />
     <Compile Include="Appium\Enums\MobileCapabilityType.cs" />
     <Compile Include="Appium\Android\Enums\ConnectionType.cs" />
+    <Compile Include="Appium\Enums\WindowsKeyCodes.cs" />
```
Comment (TikhomirovSergey) [inline appium-dotnet-driver/appium-dotnet-driver.csproj]: I think it is better to create a folder in `Appium\Windows\` and name it `Enums` and then put the file to this folder

**Comment by TikhomirovSergey** [comment]
I hope that the Windoiws Mobile-specific searching will be provided ASAP on the server-side

**Comment by jonstoneman** [comment]
Thanks for taking a look @jlipps and @TikhomirovSergey.  I've addressed your comments but the `git rebase -i` I did to get rid of the merges seems to have changed the IDs of the commits I'd merged in from `appium/appium-dotnet-driver`.  Is that expected?  Will they be discarded if you merge in this PR?  If not, I'll roll back but could do with some guidance on how to remove the merge commits cleanly.

**Comment by TikhomirovSergey** [comment]
@jonstoneman 
No. It is not expected. It seems you used a fork with very old code.
Please close this and open a new one which will contain only required changes.

One more question. Does it supports the searching by accesssibility Id?  Could you provide an additional test?

**Comment by yodurr** [comment]
Yes WinAppDriver supports searching by Accessibility ID.  I'll add a test.   

@TikhomirovSergey we're working on support for Win10 Mobile, for now this only supports Win10 PC.  Thanks!

**Comment by jonstoneman** [comment]
@TikhomirovSergey It's an old fork as most of the work to support Windows was done a long time ago.  I'll create a new one and re-apply the changes.

**Comment by jonstoneman** [comment]
Replaced by https://github.com/appium/appium-dotnet-driver/pull/158

**Pull request closed without merging**